lib/routes/whu/rsgis.ts

```diff
@@ -0,0 +1,290 @@
+import { Route, Data } from '@/types';
+import ofetch from '@/utils/ofetch';
+import { load, Cheerio, AnyNode } from 'cheerio';
+import { parseDate } from '@/utils/parse-date';
+import cache from '@/utils/cache';
+import { Context } from 'hono';
+
+const baseUrl = 'http://rsgis.whu.edu.cn';
+const categoryMap = {
+    index: {
+        name: '首页',
+        path: '',
+    },
+    xyxw: {
+        name: '学院新闻',
+        path: 'xyxw1',
+        sub: {
+            xyyw: {
+                name: '学院要闻',
+                path: 'xyyw2',
+            },
+            hzjl: {
+                name: '合作交流',
+                path: 'hzjl',
+            },
+            mtjj: {
+                name: '媒体聚焦',
+                path: 'mtjj',
+            },
+            xgyw: {
+                name: '学工要闻',
+                path: 'xgyw',
+            },
+        },
+    },
+    kxyj: {
+        name: '科学研究',
+        path: 'kxyj',
+        sub: {
+            xsbg: {
+                name: '学术报告',
+                path: 'xsbg',
+            },
+            xsjl: {
+                name: '学术交流',
+                path: 'xsjl',
+            },
+            kycg: {
+                name: '学术成果',
+                path: 'kycg',
+            },
+            sbxx: {
+                name: '申报信息',
+                path: 'sbxx',
+            },
+        },
+    },
+    tzgg: {
+        name: '通知公告',
+        path: 'tzgg1',
+        sub: {
+            xytz: {
+                name: '学院通知',
+                path: 'xytz',
+            },
+            jxdt: {
+                name: '教学动态',
+                path: 'jxdt',
+            },
+            xsdt: {
+                name: '学术动态',
+                path: 'xsdt',
+            },
+            rcyj: {
+                name: '人才引进',
+                path: 'rcyj',
+            },
+        },
+    },
+};
+
+/**
+ * Check whether the link is external.
+ *
+ * @param link Post link
+ * @returns Whether or not weixin post
+ */
+function checkExternal(link: string): boolean {
+    const matchWeixin = link.match(/^((http:\/\/)|(https:\/\/))?([\dA-Za-z]([\dA-Za-z-]{0,61}[\dA-Za-z])?\.)+[A-Za-z]{2,6}(\/)/);
+    return !!(matchWeixin && matchWeixin.length > 0);
+}
+
+/**
+ * Get information from a list of paired link and date.
+ *
+ * @param element
+ * @returns A list of RSS meta node.
+ */
+function parseListLinkDateItem(element: Cheerio<AnyNode>, currentUrl: string) {
+    const linkElement = element.find('a').first();
+    const title = linkElement.text();
+    const href = linkElement.attr('href');
+    if (href === undefined) {
+        throw 'Cannot get link';
```

Comment:
Please throw an [Error object](https://github.com/DIYgod/RSSHub/tree/master/lib/errors/types) instead of plain string.